File: src/foundation/java/io/karma/pda/foundation/component/ItemImage.java
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

package io.karma.pda.foundation.component;

import io.karma.pda.api.app.component.AbstractComponent;
import io.karma.pda.api.app.component.ComponentType;
import io.karma.pda.api.color.Color;
import io.karma.pda.api.color.ColorProvider;
import io.karma.pda.api.state.MutableState;
import io.karma.pda.api.state.Synchronize;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.Items;

import java.util.UUID;

/**
 * @author Alexander Hinze
 * @since 13/04/2024
 */
public class ItemImage extends AbstractComponent {
    @Synchronize
    public final MutableState<Item> item = MutableState.of(Items.APPLE);
    @Synchronize
    public final MutableState<ColorProvider> background = MutableState.of(Color.NONE);
    @Synchronize
    public final MutableState<ColorProvider> foreground = MutableState.of(Color.NONE);

    public ItemImage(final ComponentType<?> type, final UUID id) {
        super(type, id);
    }
}

File: src/composition/kotlin/io/karma/pda/composition/component/ItemImage.kt
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

@file:JvmName("ComponentExtensionsKt")
@file:JvmMultifileClass

package io.karma.pda.composition.component

import io.karma.pda.api.app.component.Container
import io.karma.pda.composition.Composable
import io.karma.pda.composition.Composer
import io.karma.pda.composition.LayoutProps
import io.karma.pda.foundation.component.DefaultComponents
import io.karma.pda.foundation.component.ItemImage

/**
 * @author Alexander Hinze
 * @since 07/06/2024
 */

@Composable
inline fun <C : Container> Composer<C>.ItemImage(
    crossinline layoutProps: LayoutProps,
    localName: String? = null,
    props: @Composable ItemImage.() -> Unit = {}
) = component(
    type = DefaultComponents.ITEM_IMAGE,
    layoutProps = layoutProps,
    localName = localName,
    props = props
)